### PR TITLE
No Update on Service put

### PR DESCRIPTION
**Pull request by adamschaub**
- [x] Remove "Dirty" metadata from documents. Rely only on changed inputs/outputs for triggering service updates. 
- [x] Fix ordering issues in state handling for enable & remove. 
  - [x] Enable set to false destroys the service (if active)
  - [x] Remove set to true deletes the service from db (if inactive or destroyed)
- [x] Update tests & replace nulldb with moc

### CODE REVIEW

File: manager/manager_test.go [hunk 1]
```diff
@@ -493,22 +484,21 @@ func TestUpdateService(t *testing.T) {
...
-				PluginName:    "a",
-				PluginVersion: "b",
-				Status:        model.ServiceStatusInactive,
-				Site:          "testsite",
-				Enabled:       model.NewBool(true),
-			},
-			expectStatus: model.ServiceStatusActive,
-			plugin:       model.Plugin{},
-		},
+	}{ /*
```
Comment (vgunning) [inline manager/manager_test.go]: Dead code?

Comment (adamschaub) [inline manager/manager_test.go]: WHOOOPS

Comment (adamschaub) [inline manager/manager_test.go]: fixed

**Comment by meganokeefe** [comment]
Reviewed + LGTM.

**Pull request merged**